javaagent-spi/src/main/java/io/opentelemetry/javaagent/spi/exporter/MetricCollector.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.javaagent.spi.exporter;
+
+import io.opentelemetry.sdk.metrics.export.MetricProducer;
+
+/**
+ * A {@link MetricCollector} acts as the bootstrap for metric exporters that use {@link
+ * MetricProducer} to consume the metrics.
+ *
+ * <p>Implementation of {@link MetricCollector} must be registered through the Java SPI framework.
+ */
+public interface MetricCollector {
```

Comment:
Why is this `Collector` and not `Exporter`?

Reply:
The name was suggested by @trask. I think we can do better. 

I am more in favor of removing this interface and use https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/1273

Reply:
You mean removing all individual SPIs and having one single SPI `OpenTelemetrySdkCustomizer`?

Reply:
Not really, The #1273 does not seem feasible since the static class, we could pass all the telemetry objects but that does not seem right to me.

I have renamed the class to 
```java
public interface MetricProducerCustomizer {

  /**
   * Initialize the metric collector.
   *
   * @param producer The metric producer
   * @param config The configuration
   */
  void configure(MetricProducer producer, Properties config);
}

the `configure` method matches the `TracerCustomizer`.
```

Reply:
@iNikem `MetricExporter` is for push-based metric exporters, there isn't a corresponding SDK interface for pull-based exporters

the name `MetricCollector` makes sense to me because they call `collectAllMetrics()` on the `MetricProducer`, as opposed to customizing the `MetricProducer`

but I'm also good with `MetricProducerCustomizer`

I think the important thing that we should only support one of these two configurations:

* A single `MetricProducerCustomizer` + zero `MetricExporterFactory`, or
* Zero `MetricProducerCustomizer` + any number of `MetricExporterFactory` (whose exporters we wrap in a single `IntervalMetricReader`)

(but this doesn't need to be in this PR, I created #1281 to track this)